pyrsistent/_checked_types.py

```diff
@@ -344,8 +344,12 @@ def _check(self, it):
             self._invariant_errors.extend(error_data)
 
         def add(self, element):
-            self._check([element])
-            self._pmap_evolver[element] = True
+            return self.update([element])
+
+        def update(self, elements):
+            elements = frozenset(elements)
+            self._check(elements)
+            self._pmap_evolver.update(dict(zip(elements, repeat(True))))
```

Comment:
I discovered `dict.from_keys(elements, True)` this morning which might work here.